### DISCUSSION TITLE
Starting a new slave gives errror - ERROR: output of '/usr/bin/dig localhost' does not match (?^i:query time:\s+([0-9.]+)\smsec.*)  at (eval 48) line 1.

### DISCUSSION THREAD

**Discussion by boonie-oet-gelaen** [oetiker/SmokePing#408] [Q&A]
I've recently installed a raspberry pi with the newest OS and installed smokeping as a slave. I've done this several times over the years and have made myself a little guide to not forget anything. Now I'm having this issue for the first time. When trying to start the slave it says this:

--
Sent data to Server and got new config in response.

**ERROR: output of '/usr/bin/dig localhost' does not match (?^i:query time:\s+([0-9.]+)\smsec.*)
 at (eval 48) line 1.**
--

I did google quite a bit and found only very old issue that did not have a solution for me.

Dig also does not give the desired response.

--
 dig localhost

; <<>> DiG 9.11.5-P4-5.1+deb10u11-Raspbian <<>> localhost
;; global options: +cmd
;; connection timed out; no servers could be reached

--

Why is smokeping checking localhost? Could I somehow ship this? Any other solution?

Just today I received a new modem and updated the hosts to be pinged. Now a slave that was running fine also gives me the same issue. Not sure how that happened.

I hope someone knows how to solve this. ChatGPT was not helping.

Thanks,
Dave

Any ideas what to do?

**Comment by Strykar** [comment]
Start by posting your master and slave config.

**Comment by boonie-oet-gelaen** [comment]
Solved... (with workaround)

For some odd reason smokeping does a "dig localhost" when starting a slave. The resolver did not respond to that. This makes SP fail to start.

Then I tested other resolvers manually with "dig localhost @8.8.8.8" and that does give a response. Even though it gives NXDOMAIN.

So I edited /etc/resolv.conf and added "nameserver 8.8.8.8" and now SP starts normally.

**I would consider this a bug. Or a design oversight. Should smokping be updated to not do this "dig localhost" test at startup?**

This place does not seem very active but it may help others that I post my workaround.

Thanks
Dave
